server.go

```diff
@@ -234,3 +237,128 @@ func (b *backend) Move(r *http.Request, dest *internal.Href, overwrite bool) (cr
 	}
 	return created, err
 }
+
+type HomeSet interface {
+	GetXMLName() xml.Name
+}
+
+type UserPrincipalBackend interface {
+	CurrentUserPrincipal(ctx context.Context) (string, error)
+	ListUserHomeSets(ctx context.Context) (map[string]HomeSet, error)
+}
+
+// RelativeResourcePath returns the path of a resource without the prefix of
+// the home set that the resource is in. If the resource is not in any known
+// home set, the original request path is returned.
+func RelativeResourcePath(r *http.Request, backend UserPrincipalBackend) (string, error) {
```

Comment:
Can we move `RelativeResourcePath` and `IsResourceRequest` to `internal/`? Or do we have good reasons to expose them?